oeps/oep-0031-arch-i18n.rst

```diff
@@ -0,0 +1,122 @@
+==============================
+OEP-0031: Internationalization
+==============================
+
+.. list-table::
+
+   * - OEP
+     - :doc:`OEP-0031 <oep-0031-arch-i18n>`
+   * - Title
+     - Internationalization
+   * - Last Modified
+     - 2019-01-07
+   * - Authors
+     -
+   * - Arbiter
+     -
+   * - Status
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2019-01-07
+   * - Review Period
+     -
+
+.. contents::
+   :local:
+   :depth: 2
+
+Motivation
+==========
+
+Internationalization (i18n) is a vital part of unlocking content for users around the globe and effectively supporting our business partners. In order to help us reuse localized components from app to app, the different parts of the edX codebase should align on a consistent handling of i18n. This document covers our approach to internationalizing new front-end applications and components written in React.
+
+In this OEP we use the terms locale and language more or less interchangeably. However, note that this document addresses primarily making code multilingual, i.e., the issue of presenting the same content in different languages. Considering ways to make it multi-regional, to allow for presenting different content for different countries, is out of scope for this OEP.
```

Comment:
I believe because this is specifically talking about i18n standard for frontend application implemented by reactjs, we should also call out those application using other frontend framework, or web pages rendered by django mako templates, as out of scope. We should make this explicit call out because some of the standards within this document, i.e. Message IDs, could also apply to those implementations.

Reply:
Will do!